### ISSUE TITLE
Improving the JSON reader to create complex CRUD

### ISSUE THREAD

**Issue by bramburn** [appzcoder/crud-generator#144]
Gents,
**First thing first.... Awesome Job Sohel**
I am currently working on a branch to allow for complex CRUD with unlimited OneToMany relationships as well as improving some of the "safety checks" (for example; checking if the class exists, etc...)

You can take a look at the development branch here: https://github.com/bramburn/crud-generator

The aim is to create a JSON file to create multiple MVCs and child MVCs. A use case would be when you have the following:

- Post
-- Attachment
-- Comments

The **Post** is the parent table, with many **Attachments** and **Comments**

It's other aim is to also have "showable fields", this means that there are some fields such as "view counts" that you don't want to be user editable but it is a nice to have generated in one go from your MVC data workflow.

This will allow you to navigate into one entry of the **Post** table to then create **Attachments** or **Comments** entry.

**Anyone interested in this project, please fork and help** :) 

Here is an example of the JSON file I am currently working on:

```
{
    "CRUD": [{
        "name": "Post",
        "route": "yes",
        "routePath": "",
        "routeGroup": "",
        "perPage": "",
        "controller_namespace": "Post\\",
        "modelName": "",
        "modelNamespace": "",
        "data": [{
            "fields": [{
                "name": "title",
                "type": "string",
                "showform": "yes"
            }, {
                "name": "content",
                "type": "text",
                "showform": "yes"
            }, {
                "name": "attachment",
                "route": "yes",
                "type": "OneToMany",
                "addFormToParent": "yes",
                "database_name": "attachments",
                "data": [{
                    "name": "attachments",
                    "route": "yes",
                    "routePath": "",
                    "routeGroup": "",
                    "perPage": "",
                    "controller_namespace": "Post\\Attachments\\",
                    "modelName": "post_attachments",
                    "modelNamespace": "",
                    "fields": [{
                        "name": "title",
                        "type": "string",
                        "showform": "yes"
                    }, {
                        "name": "file",
                        "type": "file",
                        "showform": "yes"
                    }]
                }]
            }, {
                "name": "view",
                "type": "integer",
                "showform": "no"
            }, {
                "name": "comments",
                "type": "OneToMany",
                "addFormToParent": "yes",
                "database_name": "testComments",
                "data": [{
                    "name": "comments",
                    "route": "yes",
                    "routePath": "",
                    "routeGroup": "",
                    "perPage": "",
                    "controller_namespace": "Post\\Comments\\",
                    "modelName": "post_attachments",
                    "modelNamespace": "",
                    "fields": [{
                        "name": "comment",
                        "type": "text",
                        "showform": "yes"
                    }]
                }]
            }]
        }]
    }]
}
```

**Comment by bramburn** [comment]
The other thing I want to add to this is "events" I want to add the ability to add events in the create, edit, delete functions of the classes of the controller. what are people's thought on that?

**Comment by bramburn** [comment]
So far so good, the code works well with 2 levels of CRUD generation. I have only created OneToMany relationships. 
The views still needs to be heavily modified to  include selective columns and forms to be shown when listing the project. 
It currently autogenerates the relationships in the model class "hasMany" and "BelongsToMany" which is very good and time saving in figuring that out.
I will hopefully complete this in a week's time for OneToMany relationships with the views fully functionally. If there is a need for many to many then it will not pick it up easily, and you'll still have to add the relationship parameters manually.
there were a few bugs in the original code in dealing with relationships, that was fixed.
There is also a lot of variables that had to be renamed to make sense when I was reading the code.

small todo list:

- [ ] Fix View command to work with relationship database and complexjson
- [ ] Clean up the code and remove any functions I used to trial some early data workflow
- [ ] add a guide to explain the complexjson properties.
